source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -28,17 +28,20 @@ public class TentacleClient : ITentacleClient
         readonly IClientScriptServiceV2 scriptServiceV2;
         readonly IClientFileTransferService fileTransferServiceV1;
         readonly IClientCapabilitiesServiceV2 capabilitiesServiceV2;
+        readonly TentacleClientSettings settings;
 
         public TentacleClient(
             ServiceEndPoint serviceEndPoint,
             IHalibutRuntime halibutRuntime,
             IScriptObserverBackoffStrategy scriptObserverBackOffStrategy,
             TimeSpan retryDuration,
             ITentacleClientObserver tentacleClientObserver,
-            ITentacleServiceDecorator? tentacleServicesDecorator)
+            ITentacleServiceDecorator? tentacleServicesDecorator,
+            TentacleClientSettings? settings = null)
```

Comment:
How about something more specific about RetrySettings as we alrewady have `TimeSpan retryDuration` that could move into the class as well. Also not nullable, force the caller pass in the settings